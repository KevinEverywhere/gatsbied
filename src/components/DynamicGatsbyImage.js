import React from 'react';
import GatsbyImage from 'gatsby-image';
import { withPrefix } from 'gatsby';


// From https://github.com/orangejellyfish/gatsby-dynamic-image
// A React component that takes a Gatsby File node and renders either a
// gatsby-image component or a native <img> element depending on whether or not
// the node is of type ImageSharp.
export default ({ node, ...props }) => {
  console.log(props);
  // if (node && node.childImageSharp && node.childImageSharp.fluid) {
  if (props && props.fluid) {
    return <GatsbyImage crossOrigin="anonymous" fluid={props.fluid} {...props} />;
  }

  // if (node && node.childImageSharp && node.childImageSharp.fixed) {
  if (props && props.fixed) {
    return <GatsbyImage crossOrigin="anonymous" fixed={props.fixed} {...props} />;
  }

  // eslint-disable-next-line jsx-a11y/alt-text
  if (node && node.publicURL) {
    return <img alt="Gallery item" src={withPrefix(props.src)} crossOrigin="anonymous" {...props} />;
  }

  // eslint-disable-next-line jsx-a11y/alt-text
  return <img alt="Gallery item" src={withPrefix(props.src)} crossOrigin="anonymous" {...props} />;
};

// import { withPrefix } from 'gatsby';
