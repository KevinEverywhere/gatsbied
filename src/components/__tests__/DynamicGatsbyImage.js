import React from "react"
import renderer from "react-test-renderer"
import DynamicGatsbyImage from "../DynamicGatsbyImage"

describe("DynamicGatsbyImage", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<DynamicGatsbyImage />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
