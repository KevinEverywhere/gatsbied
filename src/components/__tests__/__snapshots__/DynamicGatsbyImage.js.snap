// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicGatsbyImage renders correctly 1`] = `
<img
  alt="Gallery item"
  crossOrigin="anonymous"
  src="/undefined"
/>
`;
